src/cirq_iqm/iqm_sampler.py

```diff
@@ -180,3 +197,42 @@ def _send_circuits(
                 warnings.warn(f'Failed to abort job: {e}')
             finally:
                 sys.exit()
+
+
+@dataclass
+class ResultMetadata:
+    """Metadata for an IQM execution result.
+
+    Attributes:
+        job_id: ID of the computational job.
+        calibration_set_id: Calibration set used for this :class:`.IQMResult`.
+        request: Request made to run the job.
+    """
+
+    job_id: UUID
+    calibration_set_id: Optional[UUID]
```

Comment:
I don't see why the calset id is optional, but since it's that also in `iqm_client.Metadata` I guess it has to be that here as well.